src/gax-internal/src/observability.rs

```diff
@@ -432,4 +413,154 @@ mod tests {
+    where
+        S: Subscriber + for<'a> tracing_subscriber::registry::LookupSpan<'a>,
+    {
+        // on_new_span is called when a span is created. We record its initial attributes.
+        fn on_new_span(&self, attrs: &span::Attributes<'_>, id: &span::Id, _ctx: Context<'_, S>) {
+            let mut span_map = HashMap::new();
+            let mut visitor = TestVisitor(&mut span_map);
+            attrs.record(&mut visitor);
+            self.spans.lock().unwrap().insert(id.clone(), span_map);
+        }
+    }
+
+    // TestVisitor is a tracing::field::Visit implementation to extract attribute key-value pairs.
+    // We only expect &str and i64 types for span attributes.
+    struct TestVisitor<'a>(&'a mut HashMap<String, String>);
+
+    impl<'a> tracing::field::Visit for TestVisitor<'a> {
+        fn record_str(&mut self, field: &tracing::field::Field, value: &str) {
+            self.0.insert(field.name().to_string(), value.to_string());
+        }
+
+        fn record_i64(&mut self, field: &tracing::field::Field, value: i64) {
+            self.0.insert(field.name().to_string(), value.to_string());
+        }
+
+        fn record_debug(&mut self, field: &tracing::field::Field, value: &dyn std::fmt::Debug) {
+            unimplemented!("Unexpected field type for {:?}: {:?}", field, value);
+        }
+
+        fn record_u64(&mut self, field: &tracing::field::Field, value: u64) {
+            unimplemented!("Unexpected field type for {:?}: {:?}", field, value);
+        }
+
+        fn record_bool(&mut self, field: &tracing::field::Field, value: bool) {
+            unimplemented!("Unexpected field type for {:?}: {:?}", field, value);
+        }
+    }
+
+    #[tokio::test]
+    async fn test_create_span_attributes() {
+        let layer = TestLayer::default();
+        use tracing_subscriber::prelude::*;
+        let subscriber = tracing_subscriber::registry().with(layer.clone());
+
+        let id = tracing::subscriber::with_default(subscriber, || {
+            let request =
+                reqwest::Request::new(Method::GET, "https://example.com/test".parse().unwrap());
+            let options =
+                gax::options::internal::set_path_template(RequestOptions::default(), "/test");
+            const INFO: InstrumentationClientInfo = InstrumentationClientInfo {
+                service_name: "test.service",
+                client_version: "1.2.3",
+                client_artifact: "google-cloud-test",
+                default_host: "example.com",
+            };
+            let span_info = HttpSpanInfo::from_request(&request, &options, Some(&INFO), 1);
+            let span = span_info.create_span();
+            span.id().unwrap()
+        });
+
+        let spans = layer.spans.lock().unwrap();
+        let attributes = spans.get(&id).expect("span not found");
+
+        // Check all attributes set in create_span
+        assert_eq!(attributes.len(), 15, "Unexpected number of attributes");
```

Comment:
Can we convert attributes to a map and compare those?  Maybe something like:


```suggestion
        let want = HashMap::from_iter([
            (KEY_OTEL_NAME, "GET /test"),
            (KEY_OTEL_KIND, "Client"),
            ... 
        ].into_iter().map(|(k, v)| (k.to_string(), v.to_string()));
        assert_eq!(attributes, want);
```